lib/buildutils.js

```diff
@@ -0,0 +1,244 @@
+const fs = require('fs');
+const { exec, spawn } = require('child_process');
+const util = require('../lib/util');
+const logger = require('../lib/logger');
+
+module.exports.isTargetSpecified = function (argv) {
+  return (
+    argv !== null &&
+    argv.target !== null &&
+    (argv.target === 'lumin' ||
+      argv.target === 'android' ||
+      argv.target === 'ios')
+  );
+};
+
+module.exports.npmInstallIfNeeded = function (path, callback) {
+  if (fs.existsSync(`${path}/node_modules`)) {
+    callback();
+  } else {
+    logger.normal('npm install: installing');
+
+    const proc = spawn('npm', ['install'], {
+      cwd: `${path}`,
+      stdio: 'inherit',
+      shell: process.platform === 'win32'
+    });
+    createEmitter(proc, () => {
+      logger.green('npm install: success');
+      callback();
+    });
+    // proc.on('message', function (message) {
+    //   logger.normal(message);
+    // });
+    // proc.on('error', function (err) {
+    //   throw err;
+    // });
+    // proc.on('exit', function (code, signal) {
+    //   if (signal && signal !== null) {
+    //     throw Error(`npm install failed with signal: ${signal}`);
+    //   }
+    //   if (code && code !== 0) {
+    //     throw Error(`npm install failed with code: ${code}`);
+    //   }
+    //   logger.green('npm install: success');
+    //   callback();
+    // });
+  }
+};
+
+function createEmitter (proc, successCallback) {
+  proc.on('message', function (message) { logger.normal(message); });
+  proc.on('error', function (err) { throw err; });
+  proc.on('exit', function (code, signal) {
+    if (signal && signal !== null) {
+      throw Error(`Command failed with signal: ${signal}`);
+    }
+    if (code && code !== 0) {
+      throw Error(`Command failed with code: ${code}`);
```

Comment:
same here, please add the command we tried to execute.  I think we can get it from the `proc` object